src/renderer/components/inputs/TextAreaInput.tsx

```diff
@@ -1,18 +1,31 @@
 import { Textarea } from '@chakra-ui/react';
+import { Resizable } from 're-resizable';
 import { ChangeEvent, memo, useEffect, useState } from 'react';
+import { useContext, useContextSelector } from 'use-context-selector';
 import { useDebouncedCallback } from 'use-debounce';
+import { GlobalVolatileContext } from '../../contexts/GlobalNodeState';
+import { SettingsContext } from '../../contexts/SettingsContext';
+import { useMemoArray } from '../../hooks/useMemo';
 import { InputProps } from './props';
 
 interface TextAreaInputProps extends InputProps {
     resizable: boolean;
 }
 
 export const TextAreaInput = memo(
-    ({ label, inputId, useInputData, isLocked, resizable }: TextAreaInputProps) => {
+    ({ label, inputId, useInputData, useInputSize, isLocked, resizable }: TextAreaInputProps) => {
+        const zoom = useContextSelector(GlobalVolatileContext, (c) => c.zoom);
+        const { useSnapToGrid } = useContext(SettingsContext);
+        const [isSnapToGrid, , snapToGridAmount] = useSnapToGrid;
+
         const [input, setInput] = useInputData<string>(inputId);
+        const [size, setSize] = useInputSize(inputId);
         const [tempText, setTempText] = useState('');
 
         useEffect(() => {
+            if (size.width === 0 && size.height === 0) {
```

Comment:
We need to talk about the default size being 0;0. You are essentially treating 0;0 as a "size not defined" state. If you used `undefined` as the "size not defined" state, then you could get rid of the 0;0 default size completely. Even better, TypeScript would then be able to verify that you accounted for this state everywhere.